src/uipath/_utils/_auth.py

```diff
@@ -36,28 +36,42 @@ def update_env_file(env_contents):
         f.writelines(lines)
 
 
+def _has_valid_client_credentials(
+    client_id: Optional[str], client_secret: Optional[str]
+) -> bool:
+    if bool(client_id) != bool(client_secret):
+        raise ValueError(
+            "Both client_id and client_secret must be provided together for Client Credentials Authentication."
+        )
+    if client_id and client_secret:
+        return True
+    return False
```

Comment:
nitpick: this can be simplified
```suggestion
   return bool(client_id and client_secret)
```